### PR TITLE
[webgui] Provide RAttrBase::CollectDefaults method

### PR DESCRIPTION
**Pull request by linev**
Let scan any derived from `RAttrBase` class for all members, which are RAttr classes and collect all default values for them.
Before was necessary to do this by hand in `R__ATTR_CLASS` macro.
Also enforce naming convention that names of members of RAttrLine should ends with underscore symbol.

### CODE REVIEW

**Comment by linev** [comment]
@phsft-bot build

File: graf2d/gpadv7/src/RAttrBase.cxx [hunk 1]
```diff
@@ -182,3 +187,35 @@ void ROOT::Experimental::RAttrBase::Clear()
...
+
+   const std::type_info &info = typeid(*this);
+   auto thisClass = TClass::GetClass(info);
+   auto baseClass = TClass::GetClass<ROOT::Experimental::RAttrBase>();
+   if (thisClass && baseClass) {
+
+      TIter iter(thisClass->GetListOfDataMembers());
+      TObject *member = nullptr;
+      while ((member = iter()) != nullptr) {
+         TDataMember *data_member = dynamic_cast<TDataMember *> (member);
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/src/RAttrBase.cxx:209]:
```suggestion
      for (auto data_member: TRangeDynCast<TDataMember>(thisClass->GetListOfDataMembers())) {
```

File: graf2d/primitivesv7/inc/ROOT/RLine.hxx [hunk 2]
```diff
@@ -26,9 +26,9 @@ namespace Experimental {
 
 class RLine : public RDrawable {
 
-   RPadPos fP1;                            ///< line begin
-   RPadPos fP2;                            ///< line end
-   RAttrLine  fAttrLine{this, "line_"};    ///<! line attributes
+   RPadPos    fP1;                        ///< line begin
+   RPadPos    fP2;                        ///< line end
+   RAttrLine  fAttrLine{this, "line"};    ///<! line attributes
```
Comment (Axel-Naumann) [inline graf2d/primitivesv7/inc/ROOT/RLine.hxx:31]: IIRC the intention was to not line up member names in v7 anymore; you should be able to simply apply .clang_format.

**Review by Axel-Naumann — APPROVED** [review]
Some small improvements.

Also, could we - in a subsequent PR - use the member name to determine the attribute title? I.e. `fFooAbc` => "foo_abc", `fAttrBarXyz` => "bar_xyz"?

File: graf2d/gpadv7/src/RAttrBase.cxx (hunk 1 above)
Comment (linev) [reply graf2d/gpadv7/src/RAttrBase.cxx]: Done

File: graf2d/primitivesv7/inc/ROOT/RLine.hxx (hunk 2 above)
Comment (linev) [reply graf2d/primitivesv7/inc/ROOT/RLine.hxx]: I apply clang-format for all headers in primitivesv7

File: graf2d/gpadv7/src/RAttrBase.cxx (hunk 1 above)
Comment (linev) [reply graf2d/gpadv7/src/RAttrBase.cxx]: > Also, could we - in a subsequent PR - use the member name to determine the attribute title? I.e. fFooAbc => "foo_abc", fAttrBarXyz => "bar_xyz"?

https://github.com/linev/root/commit/e63b70015f1865c036ad4d998810ba41d364d8af

Implementation is simple, but super inefficient . For each instance of any attribute class each data member will try to locate - "that is my symbolic name?". And at this moment you need to access class (`const ROOT::Experimental::RAttrBase *prnt` in commit). 

Possible way of optimization - extend defaults list and store there offset of each data member. 
In this case member name can be extracted via defaults list faster - but anyway it introduces a large overhead during object construction. 

Not sure if we want this?


In contrary, list of defaults created only when used for the first time

Comment (Axel-Naumann) [reply graf2d/gpadv7/src/RAttrBase.cxx]: OK thanks, makes sense!

**Pull request merged**